coord/src/rmr/redis_cluster.c

```diff
@@ -82,15 +82,22 @@ MRClusterTopology *RedisCluster_GetTopology(RedisModuleCtx *ctx) {
       size_t hostlen, idlen;
       const char *host =
           RedisModule_CallReplyStringPtr(RedisModule_CallReplyArrayElement(nd, 0), &hostlen);
-      long long port = RedisModule_CallReplyInteger(RedisModule_CallReplyArrayElement(nd, 1));
       const char *id =
-          RedisModule_CallReplyStringPtr(RedisModule_CallReplyArrayElement(nd, 2), &idlen);
+          RedisModule_CallReplyStringPtr(RedisModule_CallReplyArrayElement(nd, 2), &idlen);
+
+      const char *id_str = rm_strndup(id, idlen);
+
+      // We need to get the port using the `RedisModule_GetClusterNodeInfo` API because on 7.2
+      // invoking `cluster slot` from RM_Call will always return the none tls port.
+      // For for information refer to: https://github.com/redis/redis/pull/12233
+      long long port = 0;
+      RedisModule_GetClusterNodeInfo(ctx, id_str, NULL, NULL, &port, NULL);
 
       MRClusterNode node = {
           .endpoint =
               (MREndpoint){
                   .host = rm_strndup(host, hostlen), .port = port, .auth = (clusterConfig.globalPass ? rm_strdup(clusterConfig.globalPass) : NULL) , .unixSock = NULL},
-          .id = rm_strndup(id, idlen),
+          .id = id_str,
```

Comment:
Should also adjust https://github.com/RediSearch/RediSearch/blob/0c5fe146b25ca4a175f9ef1d6e42f6980388818b/coord/src/rmr/cluster.c#L334
(no need to free)

Reply:
Oh OK - looks OK